### ISSUE TITLE
[BUG] Can't warm restart docker installed Rancher due to webhook conflicts

### ISSUE THREAD

**Issue by Oats87** [rancher/rancher#40335]
**Rancher Server Setup**
- Rancher version: v2.7.1, v2.7-head
- Installation option (Docker install/Helm Chart): Docker install

**Information about the Cluster**
- Kubernetes version: whatever is the K3s version we include

**Describe the bug**
On warm restarts of a Rancher docker-install, Rancher will not start due to a fatal error stemming from the rancher-webhook being unavailable.

**To Reproduce**
Run Rancher as a docker installation
Stop the rancher container
Start the rancher container

**Result**
Rancher will fail to start with logs like:
```
[root@cannon01 ~]# docker logs -f rancher39994 2>&1
INFO: Running k3s server --cluster-init --cluster-reset
2023/01/26 20:17:14 [TRACE] Loglevel set to [trace]
2023/01/26 20:17:14 [INFO] Rancher version 2c236d7b2-dirty (2c236d7b2) is starting
2023/01/26 20:17:14 [INFO] Rancher arguments {ACMEDomains:[] AddLocal:true Embedded:false BindHost: HTTPListenPort:80 HTTPSListenPort:443 K8sMode:auto Debug:false Trace:true NoCACerts:false AuditLogPath:/var/log/auditlog/rancher-api-audit.log AuditLogMaxage:10 AuditLogMaxsize:100 AuditLogMaxbackup:10 AuditLevel:0 Features: ClusterRegistry:}
2023/01/26 20:17:14 [INFO] Listening on /tmp/log.sock
2023/01/26 20:17:14 [INFO] Waiting for server to become available: Get "https://127.0.0.1:6444/version?timeout=15m0s": dial tcp 127.0.0.1:6444: connect: connection refused
2023/01/26 20:17:16 [INFO] Waiting for server to become available: Get "https://127.0.0.1:6444/version?timeout=15m0s": dial tcp 127.0.0.1:6444: connect: connection refused
2023/01/26 20:17:18 [INFO] Waiting for server to become available: Get "https://127.0.0.1:6444/version?timeout=15m0s": dial tcp 127.0.0.1:6444: connect: connection refused
2023/01/26 20:17:20 [INFO] Waiting for server to become available: Get "https://127.0.0.1:6444/version?timeout=15m0s": dial tcp 127.0.0.1:6444: connect: connection refused
2023/01/26 20:17:40 [FATAL] 2 errors occurred:
	* Internal error occurred: failed calling webhook "rancher.cattle.io.clusters.management.cattle.io": failed to call webhook: Post "[https://rancher-webhook.cattle-system.svc:443/v1/webhook/mutation/clusters.management.cattle.io?timeout=10s](https://rancher-webhook.cattle-system.svc/v1/webhook/mutation/clusters.management.cattle.io?timeout=10s)": proxy error from 127.0.0.1:6443 while dialing 10.42.0.10:9443, code 503: 503 Service Unavailable
	* Internal error occurred: failed calling webhook "rancher.cattle.io.clusters.management.cattle.io": failed to call webhook: Post "[https://rancher-webhook.cattle-system.svc:443/v1/webhook/mutation/clusters.management.cattle.io?timeout=10s](https://rancher-webhook.cattle-system.svc/v1/webhook/mutation/clusters.management.cattle.io?timeout=10s)": context deadline exceeded
```

**Expected Result**
Rancher should restart 

**Workaround**
While the Rancher container is starting, exec into it and run 
```
kubectl delete validatingwebhookconfiguration rancher.cattle.io && kubectl delete mutatingwebhookconfiguration rancher.cattle.io
```
repeatedly until the request goes through, and Rancher starts. Once rancher starts, stop doing this so it can recreate the webhook configurations.

**Comment by markusewalker** [comment]
**QA TEST PLAN**

Scenarios
|Scenario|Test Case|
|---|---|
|1|Deploy a Docker install of Rancher, stop and immediately start the container and validate that Rancher is still accessible|

**Comment by markusewalker** [comment]
Validated that this is addressed in `v2.7-head` https://github.com/rancher/rancher/commit/3a423679f0c42a02632736ede6bcc96ad7739804.

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/3a423679f0c42a02632736ede6bcc96ad7739804
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|Deploy a Docker install of Rancher, stop and immediately start the container and validate that Rancher is still accessible|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. In a client machine, installed Rancher via Docker, with `--trace` enabled.
2. Ran `docker stop <container name> && docker start <container name>` to restart Docker.
3. Ran `docker ps` and validated that the container is back up and the `STATUS` does not have any restarts.
4. Ran `docker logs -f <container name> 2>&1` and ensured that there is no error upon restarting Rancher.
5. In a browser, accessed Rancher and validated that I can reach the home page.
